test-data/unit/check-isinstance.test

```diff
@@ -1770,3 +1770,13 @@ def narrow_any_to_str_then_reassign_to_int() -> None:
         reveal_type(v)  # E: Revealed type is 'Any'
 
 [builtins fixtures/isinstance.pyi]
+
+[case testIsinstanceWidensWithAnyArg]
```

Comment:
I think this is a good fix, but could you please add few more tests:
* With an error and ``# type: ignore``
* With an "un-imported" ``Any``
* With an initial type that is not a union